subsys/bluetooth/mesh/rpl.c

```diff
@@ -290,7 +290,7 @@ BT_MESH_SETTINGS_DEFINE(rpl, "RPL", rpl_set);
 
 static void store_rpl(struct bt_mesh_rpl *entry)
 {
-	struct rpl_val rpl;
+	struct rpl_val rpl = {0};
```

Comment:
Designated initializers could be used instead.

Reply:
Designated initialization for unused part of bitfield?

Reply:
Perhaps the unused area should have a name to be initialized to 0 when using designated initializer:
```
struct rpl_val {
	uint32_t seq:24,
	      old_iv:1,
	      unused:7;
};
```

Reply:
Perhaps, but Zephyr's mesh doesn't follow such design idea.

Reply:
I do not have objections. It is worth to do as a separate PR with redesigning all bitfields within ble mesh implementation to not mix bugfix and refactoring.

Reply:
It makes clear sense to do this here. You just need to add one bit fields that does nothing to but solves the issue.